### PR TITLE
LPS-141479 When searching for a Workflow, the search returns past versions of Workflow's title

### PR DESCRIPTION
**Pull request by michaelmdelcavalcanti**
LPS: [LPS-141479](https://issues.liferay.com/browse/LPS-141479).
I created a new filter that checks if the found version is the most current version, that is, I check if the version of the found version is equal to the version that is stored in the definition, because the definition always stores the most updated version.

### CODE REVIEW

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/java/com/liferay/portal/workflow/kaleo/designer/web/internal/util/filter/KaleoDefinitionVersionVerifyVersionPredicate.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.workflow.kaleo.model.KaleoDefinition;
+import com.liferay.portal.workflow.kaleo.model.KaleoDefinitionVersion;
+
+import java.util.function.Predicate;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class KaleoDefinitionVersionVerifyVersionPredicate
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/java/com/liferay/portal/workflow/kaleo/designer/web/internal/util/filter/KaleoDefinitionVersionVerifyVersionPredicate.java:29]: Hi @michaelmdelcavalcanti, what do you think about change the name of this class to  `KaleoDefinitionVersionVersionPredicate`?

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/java/com/liferay/portal/workflow/kaleo/designer/web/internal/util/filter/KaleoDefinitionVersionVerifyVersionPredicate.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	implements Predicate<KaleoDefinitionVersion> {
+
+	@Override
+	public boolean test(KaleoDefinitionVersion kaleoDefinitionVersion) {
+		try {
+			KaleoDefinition kaleoDefinition =
+				kaleoDefinitionVersion.getKaleoDefinition();
+
+			if (kaleoDefinition.getVersion() == GetterUtil.getFloat(
```
Comment (seltong) [inline modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/java/com/liferay/portal/workflow/kaleo/designer/web/internal/util/filter/KaleoDefinitionVersionVerifyVersionPredicate.java:38]: Hey @michaelmdelcavalcanti, what do you think about reversing the order of this condition?

Something like this:
```
if (GetterUtil.getFloat(kaleoDefinitionVersion.getVersion()) ==
		kaleoDefinition.getVersion()) {

	return true;
}
```

**Review by seltong — CHANGES_REQUESTED** [review]

**Comment by michaelmdelcavalcanti** [comment]
@seltong, Ok, I will apply your suggestions.

**Comment by michaelmdelcavalcanti** [comment]
ci:test:sf

**Comment by michaelmdelcavalcanti** [comment]
ci:test:relevant

**Comment by seltong** [comment]
LGTM! :+1:  :tada:

**Review by seltong — APPROVED** [review]
LGTM! :+1: :tada:

**Comment by seltong** [comment]
ci:forward

**Pull request closed without merging**